### PR TITLE
Issue 11

### PR DESCRIPTION
**Pull request by wkozmaNTIA**

### CODE REVIEW

**Review by jhazentia — APPROVED** [review]
Looks good to me!

**Comment by sergebdt** [comment]
I will also have a look in more details in the next 2 days. Will let you know.

**Comment by wkozmaNTIA** [comment]
@sbdt-google I have a technical description of this change in a document that we are going to post here to provide details.  Its currently circulating internally.  That will help provide details pertaining to this code update.

**Comment by sergebdt** [comment]
> @sbdt-google I have a technical description of this change in a document that we are going to post here to provide details. Its currently circulating internally. That will help provide details pertaining to this code update.

That would be great if you can provide to me (sbdt@google.com), as we are trying to assess the extent it impacts the Hybrid model in Winnforum, and if/when SAS should be updated accordingly.

**Comment by sergebdt** [comment]
Hi, I have looked in more details on the fix. It is good, with possibly just a small improvement to be made.
See the analysis that is done in the new document uploaded in #11

**Comment by wkozmaNTIA** [comment]
@sbdt-google  I did a small amount of refactoring to make it more clear (explicit) where the clamps are coming into play in the computation of the correction factor.  Curve A and C are now bounding.  Take a look.

**Comment by sergebdt** [comment]
LGTM.
I updated the analysis PDF - see the doc v3 in the issue #11

File: src/IsolatedRidgeCorrectionFactor.cpp [hunk 1]
```diff
@@ -16,25 +16,47 @@ double IsolatedRidgeCorrectionFactor(double d1_hzn__km, double d2_hzn__km, doubl
...
+    // points from Figure 31, Okumura, at corresponding d2 distances
     double curve_data[3][9] =
             { {  4.0, -13.0, -17.5, -17.5, -15.0, -12.5, -10.0, -8.0, -6.0 },   // C curve : d1 <= 15 km
               { 12.0,  -8.5, -13.0, -12.0, -10.0,  -8.0,  -6.5, -5.0, -4.0 },   // B curve : d1 <= 30 km
               { 20.0,  -4.0,  -6.5,  -6.0,  -4.5,  -3.5,  -2.5, -2.0, -1.0 } }; // A curve : d1 <= 60 km
 
     // normalized ridge height factor
-    double alpha = sqrt(h_edge__meter / 200.0); // Eq 1, Okumura, alpha = 0.07 * sqrt(h)
+    // Eq 1, Okumura, alpha = 0.07 * sqrt(h)
+    double alpha = sqrt(h_edge__meter / 200.0);
```
Comment (jhazentia) [inline src/IsolatedRidgeCorrectionFactor.cpp:28]: May be helpful to add comment to indicate height is being normalized to 200 meters and that 0.07 approx equals square root of 1/200

File: src/IsolatedRidgeCorrectionFactor.cpp [hunk 2]
```diff
@@ -16,25 +16,47 @@ double IsolatedRidgeCorrectionFactor(double d1_hzn__km, double d2_hzn__km, doubl
...
 
+    // identify the d1 curve set to use
     int id1 = 0;
     if (d1_hzn__km >= d_1__km[1])
         id1 = 1;
-    int id2 = 0;
 
+    // select the first d2 curve distance that is <= to actual path d2 distance
```
Comment (jhazentia) [inline src/IsolatedRidgeCorrectionFactor.cpp:35]: Comment might be slightly confusing. Would it be more correct to say something like the following?
`select the first d2 curve distance that is directly below or equal to actual path d2 distance`

**Review by jhazentia — APPROVED** [review]
Looks good! The refactoring makes things more clear. See previous minor comments.

File: src/IsolatedRidgeCorrectionFactor.cpp (hunk 2 above)
Comment (wkozmaNTIA) [reply src/IsolatedRidgeCorrectionFactor.cpp:35]: the term "below" is problematic, as Figure 31 in the paper has plots the curves (from top to bottom) A,B,C, while the data array in the code contains the curve data as C,B,A, causing an ambiguity.  I'm going to leave the comment it as-is for now.

**Comment by sergebdt** [comment]
Thanks -- FYI I have created corresponding code reporting into Winnforum
there:
https://github.com/Wireless-Innovation-Forum/Spectrum-Access-System/pull/895

**Pull request merged**